DOC-1269 Clarify SSO with Entra ID (#392)
redpanda-data/cloud-docs: modules/security/pages/cloud-authentication.adoc

= Authentication
:description: Learn about Redpanda Cloud authentication.
:page-aliases: deploy:deployment-option/cloud/security/cloud-authentication.adoc

> :page-toclevels: 3

Reviewer: do we use this frontmatter for other pages?

Author: This is so we render headings at the ==== level in the right side nav (here, so we can see the **Integrate with Entra ID** heading). Yes, we do it in other pages; e.g., [here](https://docs.redpanda.com/redpanda-cloud/reference/properties/cluster-properties/)

==== Integrate with Microsoft Entra ID

To integrate with Entra ID, follow the https://learn.microsoft.com/en-us/entra/identity-platform/v2-protocols-oidc[Microsoft documentation^] to create an OIDC enterprise (Web) application:

> . In the https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id[Entra App Gallery^], on the Create your own application page: 

Reviewer 2: The AI bot feedback was accurate, it is not App Gallery but App Registrations https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app

Author: incorporated coderabbit's suggestions!

==== Integrate with Microsoft Entra ID

To integrate with Microsoft Entra ID, create a Web application registration that uses the OIDC Authorization Code flow with PKCE:

. In the https://entra.microsoft.com/[Microsoft Entra admin center^], go to *App registrations* and click *New registration*.
.. Name: `Redpanda Cloud`.
.. Supported account types: *Accounts in this organizational directory only (Single tenant)*.
.. Redirect URI: select *Web*, and paste the Callback URL from Redpanda Cloud. To find the Callback URL, go to *Users* > *Single sign-on* in Redpanda Cloud and click *Add connection*. Copy the *Callback URL*.
+

> IMPORTANT: The platform type must be *Web*. Because Redpanda Cloud uses the OIDC Authorization Code flow with PKCE and a server-side callback, the app must be configured as Web (not SPA or mobile).

Reviewer: this is the kind of thing that we should use doc-detective to help us. Especially microsoft as they change their UI a lot

.. Click *Register*.
. After registration, a corresponding Enterprise application (service principal) appears under *Enterprise applications*. If your organization restricts access, assign users/groups to this Enterprise application to allow access to Redpanda Cloud.
. On the application registration for Redpanda Cloud, click *Endpoints* and copy the *OpenID Connect metadata document* URL.
. In Redpanda Cloud, on the *Users*: *Single sign-on* page, paste that endpoint address into the *Discovery URI* field. Then, complete the SSO configuration:
.. For *Client ID*, copy and paste the *Application (client) ID* from the Azure app for Redpanda Cloud.
.. For *Client secret*, copy and paste the secret you get from adding a client secret on the Certificates & secrets page for the Azure app for Redpanda Cloud.
.. For *Realm*, enter your Microsoft Entra ID tenant domain name.
.. Click *Save*.
.. On the Redpanda Cloud Users: Single sign-on page, edit your new Entra ID connection to enable single sign-on. 
+
Users with an email address with that realm (domain) can now access your Redpanda Cloud account. 
+
[NOTE]
====

> * No additional claims required: Do not add optional claims or group claims to ID/Access tokens. Redpanda Cloud only relies on standard OIDC claims provided by your IdP.

Reviewer: also why do we say "do not add"? If the user adds it can he run into errors?

> * No additional claims required: Do not add optional claims or group claims to ID/Access tokens. Redpanda Cloud only relies on standard OIDC claims provided by your IdP.

Reviewer: ```suggestion
* No additional claims required: You don't need to add optional claims or group claims to ID/Access tokens. Redpanda Cloud only relies on standard OIDC claims provided by your IdP.
```

Other suggestion based on L68 if there are no errors associated with the operation

> * No additional claims required: Do not add optional claims or group claims to ID/Access tokens. Redpanda Cloud only relies on standard OIDC claims provided by your IdP.

Author: incorporated this!
